internal/doc/butane.yaml

```
@@ -326,6 +326,15 @@ root:
         - name: luks
           desc: describes the clevis configuration for encrypting the root filesystem.
           children:
+            - name: device
+              transforms:
+                - regex: $
+                  replacement: the whole-disk device (not partitions), referenced by their absolute path. A device must be start with `/dev/dasd` for `s390x-eckd` or `/dev/sd` for `s390x-zfcp` layouts.
```

Minor: to match better the phrasing used in this doc:

```suggestion
                  replacement: the whole-disk device (not partitions), referenced by their absolute path. Must start with `/dev/dasd` for `s390x-eckd` layout or `/dev/sd` for `s390x-zfcp` layouts.
```

---

The default unit test under butane ./test works Ok. 
For the layout testing I build the butane from the git branch s390x-layout then run on each layouts. 
example. s390x-zfcp, s390x-virt and s390x-eckd.

Provide with right config.

```
[root@linux butane]# cat /tmp/test.bu && ./bin/s390x/butane --pretty /tmp/test.bu -o /tmp/test.yaml && cat /tmp/test.yaml
variant: fcos 
version: 1.6.0-experimental 
boot_device:
  #layout: s390x-eckd 
  layout: s390x-zfcp 
  #layout: s390x-virt
  #layout: x86_64 
  #layout: ppc64le 
  #layout: aarch64
  luks: 
    device: /dev/sda 
    tang: 
      - url: http://12.23.21.8:7500 
        thumbprint: RJTHJG0yv54BcZ_nIfpTbQ 
  #mirror:
  #  devices:
  #    - /dev/sdb
  #    - /dev/sdc

```
Output:

```
{
  "ignition": {
    "version": "3.5.0-experimental"
  },
  "storage": {
    "filesystems": [
      {
        "device": "/dev/mapper/root",
        "format": "xfs",
        "label": "root",
        "wipeFilesystem": true
      }
    ],
    "luks": [
      {
        "clevis": {
          "tang": [
            {
              "thumbprint": "RJTHJG0yv54BcZ_nIfpTbQ",
              "url": "http://12.23.21.8:7500"
            }
          ]
        },
        "device": "/dev/sda4",
        "label": "luks-root",
        "name": "root",
        "wipeVolume": true
      }
    ]
  }
}
```
Fails on wrong device:

```
[root@linux  butane]# vim /tmp/test.bu 
[root@linux  butane]# cat /tmp/test.bu && ./bin/s390x/butane --pretty /tmp/test.bu -o /tmp/test.yaml && cat /tmp/test.yaml
variant: fcos 
version: 1.6.0-experimental 
boot_device:
  #layout: s390x-eckd 
  layout: s390x-zfcp 
  #layout: s390x-virt
  #layout: x86_64 
  #layout: ppc64le 
  #layout: aarch64
  luks: 
    device: /dev/ 
    tang: 
      - url: http://12.23.21.8:7500 
        thumbprint: RJTHJG0yv54BcZ_nIfpTbQ 
  #mirror:
  #  devices:
  #    - /dev/sdb
  #    - /dev/sdc

error at $.boot_device.s390x-zfcp, line 5 col 3: device name must start with /dev/dasd on s390x-eckd layout or /dev/sd on s390x-zfcp layout
Error translating config: source config is invalid
```

Fails on not providing the `device`:

```
[root@linux butane]# vim /tmp/test.bu 
[root@linux  butane]# cat /tmp/test.bu && ./bin/s390x/butane --pretty /tmp/test.bu -o /tmp/test.yaml && cat /tmp/test.yaml
variant: fcos 
version: 1.6.0-experimental 
boot_device:
  #layout: s390x-eckd 
  layout: s390x-zfcp 
  #layout: s390x-virt
  #layout: x86_64 
  #layout: ppc64le 
  #layout: aarch64
  luks: 
    #device: /dev/ 
    tang: 
      - url: http://12.23.21.8:7500 
        thumbprint: RJTHJG0yv54BcZ_nIfpTbQ 
  #mirror:
  #  devices:
  #    - /dev/sdb
  #    - /dev/sdc

error at $.boot_device.s390x-zfcp, line 5 col 3: device is required for layouts: s390x-eckd, s390x-zfcp
Error translating config: source config is invalid
```

Fails on Mirror:

```
[root@linux butane]# vim /tmp/test.bu 
[root@linux  butane]# cat /tmp/test.bu && ./bin/s390x/butane --pretty /tmp/test.bu -o /tmp/test.yaml && cat /tmp/test.yaml
variant: fcos 
version: 1.6.0-experimental 
boot_device:
  #layout: s390x-eckd 
  layout: s390x-zfcp 
  #layout: s390x-virt
  #layout: x86_64 
  #layout: ppc64le 
  #layout: aarch64
  luks: 
    #device: /dev/ 
    tang: 
      - url: http://12.23.21.8:7500 
        thumbprint: RJTHJG0yv54BcZ_nIfpTbQ 
  mirror:
    devices:
      - /dev/sdb
      - /dev/sdc

error at $.boot_device.s390x-zfcp, line 5 col 3: device is required for layouts: s390x-eckd, s390x-zfcp
error at $.boot_device.s390x-zfcp, line 5 col 3: mirroring not supported on layouts: s390x-eckd, s390x-zfcp, s390x-virt
Error translating config: source config is invalid
```

For s390x-virt KVM. it takes the label for device.

```
[root@linux  butane]# cat /tmp/test.bu && ./bin/s390x/butane --pretty /tmp/test.bu -o /tmp/test.yaml && cat /tmp/test.yaml
variant: fcos 
version: 1.6.0-experimental 
boot_device:
  #layout: s390x-eckd 
  #layout: s390x-zfcp 
  layout: s390x-virt
  #layout: x86_64 
  #layout: ppc64le 
  #layout: aarch64
  luks: 
    #device: /dev/ 
    tang: 
      - url: http://12.23.21.8:7500 
        thumbprint: RJTHJG0yv54BcZ_nIfpTbQ 
  #mirror:
  #  devices:
  #    - /dev/sdb
  #    - /dev/sdc

{
  "ignition": {
    "version": "3.5.0-experimental"
  },
  "storage": {
    "filesystems": [
      {
        "device": "/dev/mapper/root",
        "format": "xfs",
        "label": "root",
        "wipeFilesystem": true
      }
    ],
    "luks": [
      {
        "clevis": {
          "tang": [
            {
              "thumbprint": "RJTHJG0yv54BcZ_nIfpTbQ",
              "url": "http://12.23.21.8:7500"
            }
          ]
        },
        "device": "/dev/disk/by-partlabel/root",
        "label": "luks-root",
        "name": "root",
        "wipeVolume": true
      }
    ]
  }
}

```
Fails on Mirror:

```
[root@linux  butane]# cat /tmp/test.bu && ./bin/s390x/butane --pretty /tmp/test.bu -o /tmp/test.yaml && cat /tmp/test.yaml
variant: fcos 
version: 1.6.0-experimental 
boot_device:
  #layout: s390x-eckd 
  #layout: s390x-zfcp 
  layout: s390x-virt
  #layout: x86_64 
  #layout: ppc64le 
  #layout: aarch64
  luks: 
    #device: /dev/ 
    tang: 
      - url: http://12.23.21.8:7500 
        thumbprint: RJTHJG0yv54BcZ_nIfpTbQ 
  mirror:
    devices:
      - /dev/sdb
      - /dev/sdc

error at $.boot_device.s390x-virt, line 6 col 3: mirroring not supported on layouts: s390x-eckd, s390x-zfcp, s390x-virt
Error translating config: source config is invalid
```

For s390x-eckd  :

```
[root@linux  butane]# cat /tmp/test.bu && ./bin/s390x/butane --pretty /tmp/test.bu -o /tmp/test.yaml && cat /tmp/test.yaml
variant: fcos 
version: 1.6.0-experimental 
boot_device:
  layout: s390x-eckd 
  #layout: s390x-zfcp 
  #layout: s390x-virt
  #layout: x86_64 
  #layout: ppc64le 
  #layout: aarch64
  luks: 
    device: /dev/dasdb 
    tang: 
      - url: http://12.23.21.8:7500 
        thumbprint: RJTHJG0yv54BcZ_nIfpTbQ 
  #mirror:
  #  devices:
  #    - /dev/sdb
  #    - /dev/sdc

{
  "ignition": {
    "version": "3.5.0-experimental"
  },
  "storage": {
    "filesystems": [
      {
        "device": "/dev/mapper/root",
        "format": "xfs",
        "label": "root",
        "wipeFilesystem": true
      }
    ],
    "luks": [
      {
        "clevis": {
          "tang": [
            {
              "thumbprint": "RJTHJG0yv54BcZ_nIfpTbQ",
              "url": "http://12.23.21.8:7500"
            }
          ]
        },
        "device": "/dev/dasdb2",
        "label": "luks-root",
        "name": "root",
        "wipeVolume": true
      }
    ]
  }
}
```

Fails on wrong device :

```
[root@linux butane]# cat /tmp/test.bu && ./bin/s390x/butane --pretty /tmp/test.bu -o /tmp/test.yaml && cat /tmp/test.yaml
variant: fcos 
version: 1.6.0-experimental 
boot_device:
  layout: s390x-eckd 
  #layout: s390x-zfcp 
  #layout: s390x-virt
  #layout: x86_64 
  #layout: ppc64le 
  #layout: aarch64
  luks: 
    device: /dev/sdb 
    tang: 
      - url: http://12.23.21.8:7500 
        thumbprint: RJTHJG0yv54BcZ_nIfpTbQ 
  #mirror:
  #  devices:
  #    - /dev/sdb
  #    - /dev/sdc

error at $.boot_device.s390x-eckd, line 4 col 3: device name must start with /dev/dasd on s390x-eckd layout or /dev/sd on s390x-zfcp layout
Error translating config: source config is invalid
```

Fails on Mirror:

```
[root@linux  butane]# cat /tmp/test.bu && ./bin/s390x/butane --pretty /tmp/test.bu -o /tmp/test.yaml && cat /tmp/test.yaml
variant: fcos 
version: 1.6.0-experimental 
boot_device:
  layout: s390x-eckd 
  #layout: s390x-zfcp 
  #layout: s390x-virt
  #layout: x86_64 
  #layout: ppc64le 
  #layout: aarch64
  luks: 
    device: /dev/sdb 
    tang: 
      - url: http://12.23.21.8:7500 
        thumbprint: RJTHJG0yv54BcZ_nIfpTbQ 
  mirror:
    devices:
      - /dev/sdb
      - /dev/sdc
error at $.boot_device.s390x-eckd, line 4 col 3: device name must start with /dev/dasd on s390x-eckd layout or /dev/sd on s390x-zfcp layout
error at $.boot_device.s390x-eckd, line 4 col 3: mirroring not supported on layouts: s390x-eckd, s390x-zfcp, s390x-virt
Error translating config: source config is invalid
```